types/parameterized_decimal_type.go

```diff
@@ -0,0 +1,55 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/substrait-io/substrait-go/proto"
+)
+
+type ParameterizedDecimalType struct {
+	Nullability      Nullability
+	TypeVariationRef uint32
+	Precision        IntegerParam
+	Scale            IntegerParam
+}
+
+func (*ParameterizedDecimalType) isRootRef() {}
+func (m *ParameterizedDecimalType) WithNullability(n Nullability) Type {
+	m.Nullability = n
+	return m
```

Comment:
same as previous comment, this should probably be a value receiver, not a pointer receiver

Reply:
fixed it

Reply:
this got converted back to a pointer receiver, please change it to a value receiver?

Reply:
I deliberately converted it to pointer receiver. This is to catch error of using wrong type at compile time instead of run time.
Basically, if a interface is implemented with a pointer receiver. Value receiver won't implement this. 
Since pointer of ParameterizedDecimalType implements "FuncDefArgType" so value receiver won't satisfy interface "FuncDefArgType"
I feel this is helpful if we have "FuncDefArgType" and need to switch case on "type". We can get compile error than
(sample program is [here](https://go.dev/play/p/-N-XAX8DrVm))

Let me know if this sounds ok

Reply:
The problem is that now, unlike with other types, calling `WithNullability` actually *changes* `ParameterizedDecimalType` as opposed to nearly all other cases where calling that just returns a copy with the nullability updated.

This inconsistency will be a problem for users